docs/build/wasm/transaction-fees.md

```diff
@@ -0,0 +1,147 @@
+
+```rust
+#[ink::contract]
+mod rent {
+    use ink::storage::Mapping;
+
+    #[ink(storage)]
+    pub struct Rent {
+        map: Mapping<AccountId, u32>,
+        int: u32,
+        bool: bool,
+    }
+
+    impl Rent {
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Self { map: Default::default(), int: 0, bool: false }
+        }
+
+        #[ink(message)]
+        pub fn update_32(&mut self, i: u32) {
+            self.int = i
+        }
+
+        #[ink(message)]
+        pub fn flip_bool(&mut self) {
+            self.bool = !self.bool
+        }
+
+        #[ink(message)]
+        pub fn add_mapping_entry(&mut self) {
+            let caller = self.env().caller();
+            // Insert one item to storage. fee = 1 * PricePerItem (0.04ASTAR) =
+            // Value of the mapping is a u32, 4 bytes. fee = 4 * PricePerByte (0.002ASTAR) = 0.008ASTAR
+            // Total fee = 0.408ASTAR
+            self.map.insert(caller, &1u32);
+        }
+
+        #[ink(message)]
+        pub fn remove_mapping_entry(&mut self)  {
+            let caller = self.env().caller();
+            // Clears the value at key from storage.
+            // Remove one item from storage. fee = 1 * PricePerItem (0.04ASTR) =
+            // Remove the value of the mapping u32, 4 bytes. fee = 4 * PricePerByte (0.002ASTR) = 0.008ASTR
+            // Total reserve repatriated by caller = 0.408ASTR
+            self.map.remove(caller);
+        }
+
+        #[ink(message)]
+        pub fn remove_entry_account_id(&mut self, who: AccountId)  {
+            // Clears the value at key from storage.
+            // Remove one item from storage. fee = 1 * PricePerItem (0.04ASTR) =
+            // Remove the value of the mapping u32, 4 bytes. fee = 4 * PricePerByte (0.002ASTR) = 0.008ASTR
+            // Total reserve repatriated by caller = 0.408ASTR
+            self.map.remove(who);
+        }
+    }
+}
+```
+
+#### `add_mapping_entry`
+
+The fee (balance that is reserved (moved from free user balance to contract reserved balance)) will be:
+1. Insert one item to storage. fee = 1 * `PricePerItem` (0.04ASTR)
+2. Value of the mapping is u32, 4 bytes. fee = 4 * `PricePerByte` (0.002ASTR) = 0.008ASTR
+3. Total fee = 0.408ASTR
+
+#### `remove_mapping_entry`
+
+The balance repatriated (balance that is moved from the reserve of the contract account to the user account) will be:
+1. Remove one item from storage. fee = 1 * PricePerItem (0.04ASTR)
+2. Remove the value of the mapping u32, 4 bytes. fee = 4 * PricePerByte (0.002ASTR) = 0.008ASTR
+3. Total reserve repatriated by caller = 0.408ASTR
+
+#### `remove_entry_account_id`
+
+The caller will get balance repatriated (and not the user that was first charged for, because it is transferred from account reserved balance to free balance of caller). the caller will get the 0.408ASTR.
+
+#### `flip_bool` & `update_32`
+
+It will not have rent fees because it will not store new data on-chain (only updating value).
```

Comment:
Might be worth considering making a graphics out of this example to educate our users.